### DISCUSSION TITLE
How to get raw representation (as given by hex()) from an Fxp

### DISCUSSION THREAD

**Discussion by aniongithub** [francof2a/fxpmath#37] [Q&A] [answered]
Without converting to/from string? I need to convert floats to fixed point and upload them to CUDA memory for computations. All the astype functions convert the actual value and I can't find anything that provides the fixed-point representation.

**Comment by francof2a** [comment, marked as answer]
Hello Ani, first let me convert this issue to a discussion.

I guess you need get the `int` value which represents the raw value of the fixed-point. In that case, you can use `raw` method.

```python
from fxpmath import Fxp

x = Fxp(0.25, dtype='s8.8')
x.raw() # returns 64 as raw value
```

If my assumption is wrong, please write an example with the desired output.

**Comment by aniongithub** [comment]
Thanks for the reply!

That works great for a single value, but I have a 2D numpy array of floats that I want the raw representation for. Is there a way I can get them all in a single copy operation? Iterating element by element is slow.

**Reply by francof2a** [reply]
You could convert a entire Numpy array and then get the raw value in the same way.

```python
a = np.random.random((3, 2)) # a 2D Numpy array
a_fxp_raw = Fxp(a, dtype='s8.8').raw()
```

Does this help you?

> Note: If you already have a `Fxp` variable defined, like `x` in the first example, you can modify its value and get the raw value without the need of create a new `Fxp`.

```python
a_fxp_raw = x(a).raw()
```
